### PR TITLE
Deprecate parallel_* overloads taking the label as later argument

### PR DESCRIPTION
**Pull request by masterleinad**
This is not even documented, see https://github.com/kokkos/kokkos/wiki/Kokkos::parallel_for, https://github.com/kokkos/kokkos/wiki/Kokkos::parallel_reduce, and https://github.com/kokkos/kokkos/wiki/Kokkos::parallel_scan.

### CODE REVIEW

File: core/src/Kokkos_Parallel.hpp [hunk 1]
```diff
@@ -150,12 +150,11 @@ namespace Kokkos {
...
  */
-template <class ExecPolicy, class FunctorType>
-inline void parallel_for(
-    const ExecPolicy& policy, const FunctorType& functor,
-    const std::string& str = "",
-    std::enable_if_t<Kokkos::is_execution_policy<ExecPolicy>::value>* =
-        nullptr) {
+template <class ExecPolicy, class FunctorType,
+          class Enable =
+              std::enable_if_t<Kokkos::is_execution_policy<ExecPolicy>::value>>
```
Comment (dalg24) [inline core/src/Kokkos_Parallel.hpp:155]:
```suggestion
              std::enable_if_t<is_execution_policy<ExecPolicy>::value>>
```

Comment (dalg24) [reply core/src/Kokkos_Parallel.hpp:155]: Same comment for all occurrences below

**Review by dalg24 — APPROVED** [review]
I'd prefer if you didn't qualify the traits

File: core/src/Kokkos_Parallel.hpp [hunk 2]
```diff
@@ -392,40 +409,61 @@ inline void parallel_scan(
...
-inline void parallel_scan(const std::string& str, const ExecutionPolicy& policy,
-                          const FunctorType& functor) {
-  ::Kokkos::parallel_scan(policy, functor, str);
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_3
+template <class FunctorType>
+KOKKOS_DEPRECATED_WITH_COMMENT(
+    "Use the overload taking the label as first argument instead!")
+inline void parallel_scan(const size_t work_count, const FunctorType& functor,
+                          const std::string& str) {
+  ::Kokkos::parallel_scan("", work_count, functor);
```
Comment (dalg24) [inline core/src/Kokkos_Parallel.hpp:457]:
```suggestion
  ::Kokkos::parallel_scan(str, work_count, functor);
```

File: core/src/Kokkos_Parallel.hpp [hunk 3]
```diff
@@ -443,40 +477,53 @@ inline void parallel_scan(
...
+inline void parallel_scan(
+    const ExecutionPolicy& policy, const FunctorType& functor,
+    ReturnType& return_value,
+    std::enable_if_t<is_execution_policy<ExecutionPolicy>::value>* = nullptr) {
+  ::Kokkos::parallel_scan("", policy, functor, return_value);
+}
+
+#ifdef KOKKOS_ENABLE_DISABLE_DEPRECATED_CODE_3
+template <class ExecutionPolicy, class FunctorType, class ReturnType>
+inline void parallel_scan(
```
Comment (crtrott) [inline core/src/Kokkos_Parallel.hpp:490]: Annotate with the deprecation attribute?

**Review by crtrott — CHANGES_REQUESTED** [review]

**Review by cz4rs — APPROVED** [review]
Looks good to me.

**Pull request merged**